src/types.ts

```diff
@@ -32,7 +36,11 @@ export interface ZundoOptions<TState, PartialTState = TState> {
   onSave?: onSave<TState>;
   handleSet?: (
     handleSet: StoreApi<TState>['setState'],
-  ) => StoreApi<TState>['setState'];
+  ) => (
+    pastState: PartialTState,
+    currentState: PartialTState,
+    deltaState?: Partial<PartialTState> | null,
+  ) => void;
```

Comment:
My only concern here is that this is technically a breaking type change. Have you considered (this mess of a) type signature? 


```suggestion
  ) => (
    partial: Parameters<StoreApi<TState>['setState']>[0], // or maybe PartialTState
    replace: Parameters<StoreApi<TState>['setState']>[1], // or maybe boolean | undefined
    currentState: PartialTState,
    deltaState?: Partial<PartialTState> | null,
  ) => void;
```

Reply:
I had not previously considered this, but if it is okay with you then it is okay with me! Very cleverly avoids the breaking change!

I doubt that users will ever want to access the value of `replace` in the function returned by their `handleSet` function, but they already can as is. I will update to reflect this on my next push. I'll plan on passing `undefined` as the value for "replace" given to `curriedHandleSet`, which is what the value is now.

Reply:
Let's plan on this to avoid a breaking change. Can you please add a comment in the future `replace` will likely be deprecated and removed? Thanks!